client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-highlighted-features.ts

```diff
@@ -0,0 +1,42 @@
+import {
+	FEATURE_CUSTOM_DOMAIN,
+	FEATURE_NEWSLETTER_IMPORT_SUBSCRIBERS_FREE,
+	FEATURE_UNLIMITED_SUBSCRIBERS,
+	FEATURE_PAYMENT_TRANSACTION_FEES_10,
+	FEATURE_PAYMENT_TRANSACTION_FEES_8,
+	FEATURE_PAYMENT_TRANSACTION_FEES_4,
+} from '@automattic/calypso-products';
+import type { PlansIntent } from './use-grid-plans';
+
+/**
+ * Returns an array of features that, if they exist for a
+ * particular Plan on a pricing table, will be highlighted
+ * (ie, moved to the top of the list, and bolded).
+ *
+ * If the specified features don't exist for a particular
+ * Plan, they will be ignored.
+ *
+ * The Logic for processing the highlightedFeatures
+ * lives in the adjacent usePlanFeaturesForGridPlans hook.
+ *
+ * TODO clk: move to plans data store
+ */
+const useHighlightedFeatures = (
+	intent: PlansIntent | undefined,
```

Comment:
Why `undefined`? It looks like you are casting to `null` below from use, so `undefined` never really gets here.

I would have maybe considered an object Props instead though, so we can pass any combination and order of undefined/concrete values, more easily?

```
interface Props { 
    intent?: PlansIntent | null; 
    isInSignup?: boolean 
}
```

Reply:
I originally defined intent as PlansIntent | null, but updated it because the intent param is optional in the parent component. So I was just handling the param as is. But we can just pass it as intent ?? null in the parent. And I'm fine updating to an object Props. Just pushed both changes.

Reply:
After the sequence of type changes for props and return value, TS was having some issues inferring the type for the hook as a whole, so I've pulled that out and defined the type for hook similar to how we're doing it in adjacent files.